packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -588,11 +588,17 @@ export async function generateApimPluginEnvContent(context: MigrationContext): P
     );
     // judge if apim plugin exists
     let flag_apimPlugin = false;
-    for (const obj of projectSettingsContent["components"])
-      if (obj["name"] === "apim") {
-        flag_apimPlugin = true;
-        break;
-      }
+    if (
+      Object.keys(projectSettingsContent).includes("components") &&
```

Comment:
Why not check it use `activeResourcePlugins`, like https://github.com/OfficeDev/TeamsFx/blob/dev/packages/fx-core/src/core/middleware/utils/appYmlGenerator.ts#L117.

Reply:
good suggestion